eclipse-sirius/sirius-web
# [3873] Add the SelectionDialog on the EdgeTool

# Pull request template

## General purpose
What is the main goal of this pull request?

## Project management

## Architectural decision records (ADR)

## Dependencies

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- Hooks are properly typed:

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose

### Architecture

## Review

### How to test this PR?

- Import both 
	- [TestSelectionDialogStudio.zip](https://github.com/user-attachments/files/16819404/TestSelectionDialogStudio.zip)
	- [testSelectionDialogModel.zip](https://github.com/user-attachments/files/16819405/testSelectionDialogModel.zip)
- Open the diagram
- Create an edge between Entity1 (no matter which on) and one of the Entity2 node
- Use the "Create Edge" tool (not the "Link to")
- The Selection dialog should open
- Select an Entity2 different from the one use to create the edge and perform finish
- A new edge is created between the source and the node selected in the selection dialog.
![createEdge](https://github.com/user-attachments/assets/0a23773c-6d0e-49f4-88e6-b3744dc94756)


![selectionDialog](https://github.com/user-attachments/assets/3b8297c5-d935-4397-9422-f73c87332b04)
